src/filters/filter_conversion.hpp

```diff
@@ -403,6 +404,223 @@ COMPILER_SUPRESS("-Weffc++")
+                    } catch (const std::invalid_argument &ex) {
+                        throw filter_config_exception(std::string(
+                                                              "Unable to parse lower_bound_out of float map range filter conversion filter ") +
+                                                      ex.what(),
+                                                      own_type, own_id);
+                    } catch (const std::out_of_range &ex) {
+                        throw filter_config_exception(std::string(
+                                                              "Unable to parse lower_bound_out of float map range filter conversion filter ") +
+                                                      ex.what(),
+                                                      own_type, own_id);
+                    }
+                }
+            }
+
+            if(configuration.contains("upper_bound_out")) {
+                if (!(std::string("").compare(configuration.at("upper_bound_out"))==0)) {
+                    try {
+                        this->upper_bound_out = std::stod(configuration.at("upper_bound_out"));
+                    } catch (const std::invalid_argument &ex) {
+                        throw filter_config_exception(std::string(
+                                                              "Unable to parse upper_bound_out of float map range filter conversion filter ") +
+                                                      ex.what(),
+                                                      own_type, own_id);
+                    } catch (const std::out_of_range &ex) {
+                        throw filter_config_exception(std::string(
+                                                              "Unable to parse upper_bound_out of float map range filter conversion filter ") +
+                                                      ex.what(),
+                                                      own_type, own_id);
+                    }
+                }
+            }
+        }
+
+        virtual bool receive_update_from_gui(const std::string& key, const std::string& _value) override {
+            MARK_UNUSED(key);
+            MARK_UNUSED(_value);
+            return false;
+        }
+
+        virtual void get_output_channels(channel_mapping& map, const std::string& name) override {
+            if constexpr (std::is_same<T, uint8_t>::value) {
+                map.eight_bit_channels[name + ":value"] = &output;
+            } else if constexpr (std::is_same<T, uint16_t>::value) {
+                map.sixteen_bit_channels[name + ":value"] = &output;
+            } else if constexpr (std::is_same<T, double>::value) {
+                map.float_channels[name + ":value"] = &output;
+            }
+        }
+
+        virtual void update() override {
+            double val = (*input - lower_bound_in) / (upper_bound_in - lower_bound_in) * (upper_bound_out - lower_bound_out) + lower_bound_out;
```

Comment:
should it (like it is) be possible to exceed the input range for getting values exceeding the output range?
or should we limit the input so there cant be a value outside of the output range?

Reply:
I would introduce a parameter for that setting.